src/main/java/org/broadinstitute/hellbender/Main.java

```diff
@@ -91,6 +95,29 @@ protected static void printDecoratedExceptionMessage(final PrintStream ps, final
         return packageList;
     }
 
+
+    /**
+     * Reads from the given command-line arguments, pulls out configuration options,
+     * and initializes the configuration for this instance of Main.
+     * Returns the list of arguments without the config file option and config file path.
+     * @param args The Array of command-line arguments.
+     * @return A new Array of command-line arguments with the config file option and config file path removed.
+     */
+    protected String[] parseArgsForConfigSetupAndGetNewArgs(final String[] args) {
```

Comment:
This method should be referenced in the class javadoc list of overridable methods to point out that the configuration is initialized here.

And I've just realized that overriding this method solve some of the problems for my requirements about configuration in downstream projects:

* If no parsing of the args string is done, it removes the ability to provide by the user a config file.
* It can initialize the config class(es) for a custom properties file in the class path, using `ConfigUtils.initializeConfigurationAsProperties()`

It may be worthy to document this in the javadoc for the method, to make simpler to downstream projects to know how to deal with the configuration system.